python/lsst/cp/pipe/ptc/cpPlotPtcTask.py

```diff
@@ -0,0 +1,1522 @@
+
+    plotNormalizedCovariancesNumberOfBins : `float`, optional
+        Number of bins in `plotNormalizedCovariancesNumber` function
+        (Fig. 8, 10., of Astier+19).
+    """
+
+    ConfigClass = PlotPhotonTransferCurveConfig
+    _DefaultName = "cpPlotPtc"
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+        outputs = self.run(**inputs)
+        butlerQC.put(outputs, outputRefs)
+
+    def run(self, inputPtcDataset, dummyExposure=None, camera=None):
+        """Make the plots for the PTC task.
+
+        Parameters
+        ----------
+        inputPtcDataset : `lsst.ip.isr.PhotonTransferCurveDataset`
+            Output dataset from Photon Transfer Curve task.
+        dummyExposure :  `lsst.afw.image.Exposure`
+            The exposure used to select the appropriate PTC dataset.
+        camera : `lsst.afw.cameraGeom.Camera`
+            Camera to use for camera geometry information.
+        """
+
+        if len(dummyExposure) == 0:
+            self.log.warning("No dummy exposure found.")
+
+        ptcFitType = inputPtcDataset.ptcFitType
+        self.detId = inputPtcDataset.getMetadata()["DETECTOR"]
+
+        if ptcFitType in [
+            "FULLCOVARIANCE",
+        ]:
+            figDict = self.covAstierMakeAllPlots(inputPtcDataset)
+        elif ptcFitType in ["EXPAPPROXIMATION", "POLYNOMIAL"]:
+            figDict = self._plotStandardPtc(inputPtcDataset)
+        else:
+            raise RuntimeError(
+                f"The input dataset had an invalid dataset.ptcFitType: {ptcFitType}. \n"
+                "Options: 'FULLCOVARIANCE', EXPAPPROXIMATION, or 'POLYNOMIAL'."
+            )
+
+        if self.config.doPdf:
+            filenameFull = self.config.outputDirectory + "/" + self.config.filenameRoot
+            filenameFull += f"_det{self.detId}.pdf"
+            self.makePdfFile(filenameFull, figDict)
+
+        maxNumberPlots = 12
+        if len(figDict) < maxNumberPlots:
+            for i in range(len(figDict), maxNumberPlots + 1):
+                figDict.setdefault(i, plt.figure())
+
+        return pipeBase.Struct(
+            ptcPlot1=figDict[0],
+            ptcPlot2=figDict[1],
+            ptcPlot3=figDict[2],
+            ptcPlot4=figDict[3],
+            ptcPlot5=figDict[4],
+            ptcPlot6=figDict[5],
+            ptcPlot7=figDict[6],
+            ptcPlot8=figDict[7],
+            ptcPlot9=figDict[8],
+            ptcPlot10=figDict[9],
+            ptcPlot11=figDict[10],
+            ptcPlot12=figDict[11],
+        )
+
+    def makePdfFile(self, filenameFull, figDict):
+        """Save plots in a PDF file.
+
+        Parameters
+        ----------
+        filenameFull : `str`
+            File name, including full path.
+        figDict : `dict` [`int`, `~matplotlib.figure.Figure]
+            Dictionary, keyed by an integer index, of
+            matplotlib PTC figures.
+        """
+        with PdfPages(filenameFull) as pdfPages:
```

Comment:
This may need a try/except block;  I don't know if this will run correctly on a fully virtualized/cloud/etc node.  You can probably skip it for now, but if this gets added to a more involved pipeline definition it might cause issues.

Reply:
mm, so should I just delete this PDF option? I just wanted to keep the option of having a single PDF, which was convenient before for the original script.

Reply:
I decided to delete this altogether, so it's more in line with the standards.  The user can make a PDF from a notebook using the retrieved PNGs.